helix-view/src/theme.rs

```diff
@@ -288,9 +288,76 @@ fn build_theme_values(
         highlights.push(style);
     }
 
+    let fallback_background_color = Color::Black;
+    let background_color = styles
+        .get("ui.background")
+        .map_or(fallback_background_color, |style| {
+            style.bg.unwrap_or(fallback_background_color)
+        });
+
+    let focused_cursor_keys = [
+        "ui.cursor",
+        "ui.cursor.normal",
+        "ui.cursor.insert",
+        "ui.cursor.select",
+        "ui.cursor.primary",
+        "ui.cursor.primary.normal",
+        "ui.cursor.primary.insert",
+        "ui.cursor.primary.select",
+    ];
+
+    for focused_cursor_key in focused_cursor_keys {
+        if let Some(cursor_scope) = styles.get(focused_cursor_key) {
+            let unfocused_cursor_key = focused_cursor_key.to_owned() + ".unfocused";
+            if styles.contains_key(&unfocused_cursor_key) {
+                continue;
+            }
+
+            // DIM does not work - calculate dimmed color from fg / bg mix
+
+            if let Some(dimmed_color) = cursor_scope.bg.and_then(|cursor_color| {
+                calculate_transparent_color(&background_color, &cursor_color, 100)
+            }) {
+                let cursor_unfocused_scope = cursor_scope.bg(dimmed_color);
+                styles.insert(unfocused_cursor_key.to_owned(), cursor_unfocused_scope);
+                scopes.push(unfocused_cursor_key.to_owned());
+                highlights.push(cursor_unfocused_scope);
+            }
+        }
+    }
+
     (styles, scopes, highlights)
 }
 
+fn calculate_transparent_color(
```

Comment:
Automatically figuring out a default dimmed color seems unnecessary to me. If a theme wants to take advantage of focused/unfocused it should define the `*.unfocused` keys

Reply:
Agreed. Removed it.